tests/integration/identity.go

```diff
@@ -123,25 +128,27 @@ func generateIdentity(s *state) acpIdentity.Identity {
 	return identity
 }
 
+func generateToken(s *state, ident *acpIdentity.Identity, nodeID int) string {
+	audience := getNodeAudience(s, nodeID)
+	if acpType == SourceHubACPType || audience.HasValue() {
+		err := ident.UpdateToken(
+			authTokenExpiration,
+			audience,
+			immutable.Some(s.sourcehubAddress),
+		)
+		require.NoError(s.t, err)
+		return ident.BearerToken
+	}
+	return ""
+}
+
 // getContextWithIdentity returns a context with the identity for the given reference and node index.
 // If the identity does not exist, it will be generated.
 // The identity added to the context is prepared for a request, i.e. its [Identity.BearerToken] is set.
-func getContextWithIdentity(
-	ctx context.Context,
-	s *state,
-	ref immutable.Option[identityRef],
-	nodeIndex int,
-) context.Context {
-	if !ref.HasValue() {
+func getContextWithIdentity(ctx context.Context, s *state, ref identity, nodeIndex int) context.Context {
```

Comment:
todo: Don't need to call the arg `ref` anymore specially with the `identityRef` rename.